code-challenges.test.js

```diff
@@ -22,25 +36,181 @@ const people = [
+// console.log(newArray)
+
+return newArray
+
+
+}
+
+
+
+
+
+
+
+// // // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+describe("third", () => {
+  it("returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(hodgepodge1).toContain[ 2, 0, -1, 0]
+    expect(hodgepodge2).toContain[ 2, 1, -1]
+  })
+})
+// // // a) Create a test with an expect statement using the variables provided.
 
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
+// Input -an array of mixed data
+// Output -returns an array of only the REMAINDERS of the numbers when divided by 3.
+// Psuedo code - I want make a function named third and pass in a array , I want to sort the array to push all off the numbers the the front indexes and then use the slice function and take only the numbers. After that i assigned the array with only the numbers remaining and then created const for each index so that i can munipulate them. After each indexs has been %3 they are stacked into newArray and returned.  
 
-// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
+const third = (array)=>{
+
+const grabber = array.sort()
+// console.log(grabber)
+const pop = array.slice(0,4)
+
+const divider1 = pop[0]%3
+// console.log(divider1)
+
+const divider2 = pop[1]%3
+// console.log(divider2)
+
+const divider3 = pop[2]%3
+// console.log(divider3)
+
+const divider4 = pop[3]%3
+// console.log(divider4)
+
+const newArray = []
+const collecttor = newArray.push(divider1,divider2,divider3,divider4)
+console.log(newArray)
+
+return newArray
+}
+
+
+
+
+
+// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+describe("cubie", () => {
+  it("returns the sum of all the numbers cubed", () => {
+    expect(cubie(cubeAndSum1)).toEqual(99)
+    expect(cubie(cubeAndSum2)).toEqual(1125)
+  })
+})
 // a) Create a test with an expect statement using the variables provided.
 
+// Input - Array
+// Output -returns the sum of all the numbers cubed /to the 3rd power
+// Psuedo code -create a functin named cubie and pass in a array, each array has 3 indexes for I destructured the array into each index and gave the name of NUM1-3. After that I created a new const called results which just take the index and muiltpy each index to the 3rd power seperately and then adding the results of the multiplication.
+
+
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 // b) Create the function that makes the test pass.
+
+const cubie = (array)=>{
+
+const num1 = array[0] 
```

Comment:
What is another way we can access the elements in an array and perform some logic on them without having to hard code the [0] - [2]?